freqtrade/plugins/pairlist/PercentVolumeChangePairList.py

```diff
@@ -0,0 +1,319 @@
+            since_ms = (
+                int(
+                    timeframe_to_prev_date(
+                        self._lookback_timeframe,
+                        dt_now()
+                        + timedelta(
+                            minutes=-(self._lookback_period * self._tf_in_min) - self._tf_in_min
+                        ),
+                    ).timestamp()
+                )
+                * 1000
+            )
+
+            to_ms = (
+                int(
+                    timeframe_to_prev_date(
+                        self._lookback_timeframe, dt_now() - timedelta(minutes=self._tf_in_min)
+                    ).timestamp()
+                )
+                * 1000
+            )
+
+            # todo: utc date output for starting date
+            self.log_once(
+                f"Using change range of {self._lookback_period} candles, timeframe: "
+                f"{self._lookback_timeframe}, starting from {format_ms_time(since_ms)} "
+                f"till {format_ms_time(to_ms)}",
+                logger.info,
+            )
+            needed_pairs: ListPairsWithTimeframes = [
+                (p, self._lookback_timeframe, self._def_candletype)
+                for p in [s["symbol"] for s in filtered_tickers]
+                if p not in self._pair_cache
+            ]
+
+            candles = self._exchange.refresh_ohlcv_with_cache(needed_pairs, since_ms)
+
+            for i, p in enumerate(filtered_tickers):
+                pair_candles = (
+                    candles[(p["symbol"], self._lookback_timeframe, self._def_candletype)]
+                    if (p["symbol"], self._lookback_timeframe, self._def_candletype) in candles
+                    else None
+                )
+
+                # in case of candle data calculate typical price and change for candle
+                if pair_candles is not None and not pair_candles.empty:
+                    pair_candles["rolling_volume_sum"] = (
+                        pair_candles["volume"].rolling(window=self._lookback_period).sum()
+                    )
+                    pair_candles["rolling_volume_change"] = (
+                        pair_candles["rolling_volume_sum"].pct_change() * 100
+                    )
+
+                    # ensure that a rolling sum over the lookback_period is built
+                    # if pair_candles contains more candles than lookback_period
+                    rolling_volume_change = pair_candles["rolling_volume_change"].fillna(0).iloc[-1]
+
+                    # replace change with a range change sum calculated above
+                    filtered_tickers[i]["rolling_volume_change"] = rolling_volume_change
+                    self.log_once(f"ticker {filtered_tickers[i]}", logger.info)
+                else:
+                    filtered_tickers[i]["rolling_volume_change"] = 0
+        else:
+            filtered_tickers = [v for k, v in tickers.items() if k in pairlist]
```

Comment:
This won't work - `fetch_tickers()` (the data you're using here) doesn't contain "rolling_volume_change" (your sort key).

That simply means there won't be "no candles" fallback - but it'll be more similar to pairlists like "RangeStabilityFilter" or VolatilityFilter - which are a lot simpler.

Reply:
Aah okay. Lemme work on that.

Reply:
this is no longer required post the new changes.